### PR TITLE
feat: add duckdb_library_version method and fix extension load state

### PR DESCRIPTION
**Pull request by Mause**
Used primarily for developing third party C extensions.

### CODE REVIEW

**Comment by Mytherin** [comment]
Thanks for the PR!

I'm not such a fan of having the extension call `duckdb_set_extension_loaded` itself. Since C extensions cannot be statically loaded and would need to be loaded through `ExtensionHelper::LoadExternalExtension`, could we make the call there and avoid having to expose this function/avoid forcing every extension to call this correctly?

**Comment by Mause** [comment]
Ah that's a much better spot for that yes, and explains some complaints we've had about external extensions never showing as loaded

File: src/main/extension/extension_load.cpp [hunk 1]
```diff
@@ -132,6 +132,8 @@ void ExtensionHelper::LoadExternalExtension(ClientContext &context, const string
 		throw InvalidInputException("Initialization function \"%s\" from file \"%s\" threw an exception: \"%s\"",
 		                            init_fun_name, res.filename, e.what());
 	}
+
+	DuckDB(DatabaseInstance::GetDatabase(context)).SetExtensionLoaded(extension);
```
Comment (Mytherin) [inline src/main/extension/extension_load.cpp:136]: Can we perhaps move the `SetExtensionLoaded` function to the `DatabaseInstance` instead?

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**